ocs_ci/utility/aws.py

```diff
@@ -464,3 +545,57 @@ def get_data_volumes(instance_id):
         vol for vol in volumes if vol.attachments[0]
         .get('DeleteOnTermination') is False
     ]
+
+
+def get_vpc_id_by_node_obj(aws_obj, instances):
+    """
+    This function getting vpc id by randomly selecting instances out of user aws deployment
+
+    Args:
+        aws_obj (obj): AWS() object
+        instances (dict): cluster ec2 instances objects
+
+    Returns:
+        str: vpc_id: The vpc id
+
+    """
+
+    instance_id = random.choice(list(instances.keys()))
+    vpc_id = aws_obj.get_vpc_id_by_instance_id(instance_id)
+
+    return vpc_id
+
+
+def block_availability_zone(security_group_id, instances):
+    """
+    Block all instances in randomly selected aws availability zone(az)
+    - Fetch az out of cluster instances
+    - Append security group without access permissions to all cluster nodes within same az
+
+    Args:
+        security_group_id (str): security group without access permissions
+        instances (dict): cluster ec2 instances objects
+
+    Returns:
+        list: blocked instances: all instances within the same availability zone
+
+    """
+    aws_obj = AWS()
+    instances_to_block = list()
+    instances_id = list(instances.keys())
+    random_az_selector = random.choice(instances_id)
+    random_az_selected = aws_obj.get_availability_zone_id_by_instance_id(random_az_selector)
+    logger.info(f"random_az: {random_az_selector}")
+    for instance in instances_id:
+        az = aws_obj.get_availability_zone_id_by_instance_id(instance)
+        if random_az_selected == az:
+            instances_to_block.append(instance)
+
+    logger.info(f"instances to block: {instances_to_block}")
+    aws_obj.append_security_group(security_group_id, instances_to_block)
+
+    return instances_to_block
+
+
+def restore_availability_zone():
+    pass
```

Comment:
is this part TODO ?

Reply:
yes, 
current pull request contains only test infrastructure without actual test and its tear down, to be done in next PR